sstables/generation_type.hh

```diff
@@ -33,6 +34,10 @@ public:
     explicit constexpr generation_type(int_t value) noexcept: _value(value) {}
     constexpr int_t value() const noexcept { return _value; }
 
+    explicit operator data_value() const noexcept {
+        return data_value(_value);
```

Comment:
How will this work when we support uuid? Sure data_value is a variant type and can hold all types, but the CQL statement won't work since the data type of the column is just one, not a variant\<uuid, bigint>.

I think this needs to be fixed another way:
1. change the data type in schema_registry to uuid
2. forbid using integer generations with tablets

Reply:
For object storage, we cannot use integer generations because there we will have conflicting file names in the same bucket. So object storage support must enforce uuid generations.

Reply:
> How will this work when we support uuid? Sure data_value is a variant type and can hold all types, but the CQL statement won't work since the data type of the column is just one, not a variant<uuid, bigint>.
> 
> I think this needs to be fixed another way:
> 
>     1. change the data type in schema_registry to uuid
> 
>     2. forbid using integer generations with tablets

i will implement the first step in the follow-up change which introduces the UUID based generations. regarding to the second step, i think we can keep the change in this PR, but add a TODO: so only the UUID based generation is allowed to be converted to `data_value`, so we don't have to force callers like `system_keyspace::sstables_registry_lookup_entry()` to check if the generation is UUID based or not, instead the `generation_type::operator data_value()` will do this check, and throw a runtime exception if this check fails. also, will add a check to enforce object storage's hard dependency on UUID generations.

Reply:
ok

Reply:
I think in this case, we should at least make `operator data_value` `explicit`, or even maybe a proper method. If something requires consideration, we shouldn't make it work implicitly.

Reply:
Ok, I see it is already explicit.

Reply:
We shouldn't have two identifiers for the same thing. Why do we have both entry_uuid and generation?